File: frontend/src/services/authService.js
import axiosClient from "./axiosClient";

export const login = (payload) => axiosClient.post("/auth/login", payload);
export const loginWithGoogle = (payload) => axiosClient.post("/auth/google", payload);
export const register = (payload) => axiosClient.post("/auth/register", payload);
export const sendRegisterOtp = (payload) => axiosClient.post("/auth/register/send-otp", payload);
export const forgotPassword = (payload) => axiosClient.post("/auth/forgot-password", payload);
export const resetPassword = (payload) => axiosClient.post("/auth/reset-password", payload);
export const changePassword = (payload) => axiosClient.post("/auth/change-password", payload);
export const getCurrentUser = () => axiosClient.get("/auth/me");
export const logout = (payload) => axiosClient.post("/auth/logout", payload);

File: backend/src/main/java/com/clinicmanagement/auth/AuthService.java
package com.clinicmanagement.auth;

import com.clinicmanagement.auth.dto.*;
import com.clinicmanagement.user.User;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);

    void sendRegisterOtp(RegisterOtpRequest request);

    LoginResponse login(LoginRequest request);

    LoginResponse loginWithGoogle(GoogleLoginRequest request);

    TokenResponse refreshToken(RefreshTokenRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    void changePassword(User currentUser, ChangePasswordRequest request);
}
